internal/pkg/deploy/pipeline.go

```diff
@@ -607,30 +646,123 @@ func (stg *PipelineStage) Deployments() ([]DeployAction, error) {
 	return actions, nil
 }
 
-// PreDeployments returns a list of pre-deployment actions for the pipeline stage.
-func (stg *PipelineStage) PreDeployments() ([]Step, error) {
-	return nil, nil
+// PostDeployments returns a list of post-deployment actions for the pipeline stage.
+func (stg *PipelineStage) PostDeployments() ([]PrePostDeployAction, error) {
+	if len(stg.postDeployments) == 0 {
+		return nil, nil
+	}
+
+	var prevActions []orderedRunner
+	if approval := stg.Approval(); approval != nil {
+		prevActions = append(prevActions, approval)
+	}
+	preDeployActions, err := stg.PreDeployments()
```

Comment:
probably a dumb question but: why does every one of these (`PreDeployments()`, `Deployments()`, `PostDeployments()`, and `Test()`) call `stg.PreDeployments()` 😭

Reply:
That will be refactored in a different PR, but right now it's because we don't know which of these are called. It's possible that only Pre and Post are used, so we can't count on Deployments() to account for Pre's actions in the prevAction tally